tests/metrics/test_metric_lightning.py

```diff
@@ -0,0 +1,79 @@
+import torch
+from tests.base.boring_model import BoringModel
+from pytorch_lightning.metrics import Metric
+from pytorch_lightning import Trainer
+
+
+class SumMetric(Metric):
+    def __init__(self):
+        super().__init__()
+        self.add_state("x", torch.tensor(0.0), dist_reduce_fx="sum")
+
+    def update(self, x):
+        self.x += x
+
+    def compute(self):
+        return self.x
+
+
+def test_metric_lightning(tmpdir):
+    class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.metric = SumMetric()
+            self.sum = 0.0
+
+        def training_step(self, batch, batch_idx):
+            x = batch
+            self.metric(x.sum())
+            self.sum += x.sum()
+
+            return self.step(x)
+
+        def training_epoch_end(self, outs):
+            assert torch.allclose(self.sum, self.metric.compute())
```

Comment:
now much close, lets spec tolerance, also why not exact?

Reply:
default tolerance is 1e-8, i can parameterize if you want. Can't do exact bc of floating point